opentreemap/treemap/js/src/inlineEditForm.js

```diff
@@ -174,7 +164,36 @@ exports.init = function(options) {
 
         responseStream = saveStream
             .map(getDataToSave)
-            .flatMap(update),
+            .flatMapLatest(function (data) {
+                // onSaveBefore is a function that takes form data
+                // and carries out arbitrary, potentially blocking
+                // actions before allowing the save process to
+                // continue.
+                //
+                // the return value of an onSaveBefore callback can be:
+                // * null - for non-blocking side effects and mutation
+                //          of the data object
+                // * an eventStream - for blocking side effects,
+                //                    failure cases and early exits.
+                //
+                // when providing an eventStream as the return value for
+                // onSaveBefore, it should meet the following criteria:
+                // * it should be a stream of data objects. If you are using
+                //   eventStreams to block on IO, map them to the `data`
+                //   object provided.
+                // * it should fire an event when it's ok to proceed and
```

Comment:
The phrase "fire an event" might be confusing here. If I correctly understand this new API, "It should push a new stream value..." may be a better description of the requirement.